src/JsonDB.ts

```diff
@@ -174,11 +174,23 @@ export default class JsonDB {
             if (!found || found.length < 2) {
                 return undefined
             }
-            return found[1] as T
+            return found
         }
         throw new DataError("The entry at the path (" + rootPath + ") needs to be either an Object or an Array", 12)
     }
 
+    /**
+     * Find a specific entry in an array/object
+     * @param rootPath base dataPath from where to start searching
+     * @param callback method to filter the result and find the wanted entry. Receive the entry and it's index.
+     */
+    public find<T>(rootPath: string, callback: FindCallback): T | undefined {
+        found = this.findAll(rootPath, callback)
```

Comment:
Sorry, I am quite busy right now... It would be nice of you if you put up some tests for `findAll()` Although it shouldn't fail because it was already the original `find()` function that I have expanded.

Reply:
The test are failing for the "new" find<T> method because you use an undeclared variable which is forbidden in TypeScript.

For the test, I can add it later, but I can't merge this PR until this little change is made.